### PR TITLE
Add AccountMigrationManager with OTP accounts import and export methods

### PR DESCRIPTION
**Pull request by vahancouver**
# JIRA Ticket

[SDKS-2730](https://bugster.forgerock.org/jira/browse/SDKS-2730)

# Description

[FRA][iOS] Implement Export and Import functions for OTP accounts

### CODE REVIEW

File: FRAuthenticator/FRAuthenticator/Error/AccountMigrationError.swift [hunk 1]
```diff
@@ -0,0 +1,79 @@
...
+    /// Error codes for each error enum
+    public var errorCode: Int {
+        return self.parseErrorCode()
+    }
+    
+    /// Error UserInfo
+    public var errorUserInfo: [String : Any] {
+        switch self {
+        case .invalidScheme:
+            return [NSLocalizedDescriptionKey: "Invalid schem in url given for Account Migration"]
```
Comment (rodrigoareis) [inline FRAuthenticator/FRAuthenticator/Error/AccountMigrationError.swift:70]: *Invalid scheme

File: FRAuthenticator/FRAuthenticator/Manager/AccountMigrationManager.swift [hunk 2]
```diff
@@ -0,0 +1,288 @@
...
+                let totpMechanism = TOTPMechanism(issuer: otpParameter.issuer, accountName: otpParameter.name, secret: secret, algorithm: otpParameter.algorithm.stringValue, digits: otpParameter.digits.intValue)
+                mechanisms.append(totpMechanism)
+            }
+        }
+        
+        return mechanisms
+    }
+    
+    
+    /// Decodes given OTPAauth Migration `URL` to otpauth `URL`s
```
Comment (rodrigoareis) [inline FRAuthenticator/FRAuthenticator/Manager/AccountMigrationManager.swift:65]: *Decodes given OTPAuth

File: FRAuthenticator/FRAuthenticator/Manager/AccountMigrationManager.swift [hunk 3]
```diff
@@ -0,0 +1,288 @@
...
+//  Copyright (c) 2023 ForgeRock. All rights reserved.
+//
+//  This software may be modified and distributed under the terms
+//  of the MIT license. See the LICENSE file for details.
+//
+
+
+import Foundation
+
+public struct AccountMigrationManager {
```
Comment (rodrigoareis) [inline FRAuthenticator/FRAuthenticator/Manager/AccountMigrationManager.swift:14]: Missing doc

File: FRAuthenticator/FRAuthenticator/Manager/AccountMigrationManager.swift [hunk 4]
```diff
@@ -0,0 +1,288 @@
+//
```
Comment (rodrigoareis) [inline FRAuthenticator/FRAuthenticator/Manager/AccountMigrationManager.swift:1]: We need to think about if we want to make it public or just add import and export methods on the FRClient. My concern is the easy access to the mechanism secrets the developers would have. This was not is the design, so it's something up to discussion.

**Review by spetrov — APPROVED** [review]
Looks great! 👍🏻

File: FRAuthenticator/FRAuthenticator/Manager/AccountMigrationManager.swift [hunk 5]
```diff
@@ -0,0 +1,291 @@
...
+        let mechanisms: [OathMechanism] = try decodeToMechanisms(url: url)
+        return mechanisms.compactMap { createUrl(mechanism: $0) }
+    }
+    
+    
+    /// Encodes given `OathMechanism`s into an OTPAuth Migration `URL`
+    /// - Parameter mechanisms: Array of `OathMechanism`s of types of hotp or totp
+    /// - Returns:  OTPAuth Migration `URL` with otpauth-migration scheme
+    /// - Throws: BinaryEncodingError
+    public static func encode(mechanisms: [OathMechanism]) throws -> URL? {
```
Comment (rodrigoareis) [inline FRAuthenticator/FRAuthenticator/Manager/AccountMigrationManager.swift:80]: Call out the limit of accounts to export in the doc

**Review by rodrigoareis — APPROVED** [review]
Changes looks good to me

**Review by spetrov — APPROVED** [review]

**Pull request merged**